homeassistant/components/hive.py

```diff
@@ -0,0 +1,707 @@
+                        "hive_hotwater_state" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"] == "HotWater_Mode" and
+                        "hive_hotwater_mode" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"] == "HotWater_Boost" and
+                        "hive_hotwater_boost" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"]
+                        == "Hive_Device_BatteryLevel" and
+                        "hive_thermostat_batterylevel" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"]
+                        == "Hive_Device_BatteryLevel" and
+                        "hive_sensor_batterylevel" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"] == "Hive_Device_Sensor" and
+                        "hive_active_sensor" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"]
+                        == "Hive_Device_Light_Mode" and
+                        "hive_active_light_sensor" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+                if (a_dl_device["HA_DeviceType"]
+                        == "Hive_Device_Plug_Mode" and
+                        "hive_active_plug_sensor" in config_devices):
+                    device_list_sensor.append(a_dl_device)
+
+        for a_dl_device in api_dl_climate:
+            if "HA_DeviceType" in a_dl_device:
+                if (a_dl_device["HA_DeviceType"] == "Heating" and
+                        "hive_heating" in config_devices):
+                    device_list_climate.append(a_dl_device)
+                if (a_dl_device["HA_DeviceType"] == "HotWater" and
+                        "hive_hotwater" in config_devices):
+                    device_list_climate.append(a_dl_device)
+
+        for a_dl_device in api_dl_light:
+            if "HA_DeviceType" in a_dl_device:
+                if (a_dl_device["HA_DeviceType"] == "Hive_Device_Light" and
+                        "hive_active_light" in config_devices):
+                    device_list_light.append(a_dl_device)
+
+        for a_dl_device in api_dl_switch:
+            if "HA_DeviceType" in a_dl_device:
+                if (a_dl_device["HA_DeviceType"] == "Hive_Device_Plug" and
+                        "hive_active_plug" in config_devices):
+                    device_list_switch.append(a_dl_device)
+
+    else:
+        device_list_sensor = api_dl_sensor
+        device_list_climate = api_dl_climate
+        device_list_light = api_dl_light
+        device_list_switch = api_dl_switch
+
+    global HGO

```

Comment:
No global please, we have [`hass.data[DATA_XY]`](https://home-assistant.io/developers/code_review_component/#componentplatform-communication).

Reply:
no globals now, this has been moved to hass.data[HIVE_DATA]